File: retrofithelper/src/main/java/com/sovegetables/retrofit/LoggerInterceptor.java
package com.sovegetables.retrofit;


import android.util.Log;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import okhttp3.*;
import okio.Buffer;

public class LoggerInterceptor implements Interceptor {

    private static final String TAG = "LoggerInterceptor %s";
    private static final String LINE_SEPARATOR = System.getProperty("line.separator");

    @NotNull
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        printRequest(request);
        Response response = chain.proceed(request);
        ResponseBody body = response.body();
        int code = response.code();
        Headers headers = response.headers();
        String strBody = body == null? "" : body.string();
        printResponse(strBody, headers, code);
        ResponseBody newBody = ResponseBody.create(body == null? null: body.contentType(), strBody);
        return response.newBuilder().body(newBody).build();
    }

    private void printResponse(String body, Headers headers, int code) {
        Log.d(TAG, "response code:" +  code);
        Log.d(TAG, "body:" +  body);
        Log.d(TAG, "headers:" +  headers);
    }

    private void printRequest(Request request) {
        Log.d(TAG, "url:" +  request.url());
        Log.d(TAG, "method:" +  request.method());
        Log.d(TAG, "headers:" +  request.headers());
        Log.d(TAG, "headers:" +  request.body().contentType().toString());
        Buffer buffer = new Buffer();
        try {
            request.body().writeTo(buffer);
        } catch (Exception ignored) {
        }
        Log.d(TAG, "body:" +  buffer.readByteString().utf8());
    }
}

File: retrofithelper/src/main/java/com/sovegetables/retrofit/LoggerInterceptor.kt
package com.sovegetables.retrofit

import com.sovegetables.logger.ILog
import com.sovegetables.logger.Logger
import okhttp3.*
import okio.Buffer
import java.io.IOException

class LoggerInterceptor: Interceptor{

    companion object{
        private val TAG = "LoggerInterceptor"
    }

    class InterceptorLogger: ILog() {

        override fun println(priority: LEVEL, tag: String?, msg: String?) {
            if(TAG == tag){

            }
        }

    }

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        printRequest(request)
        val response = chain.proceed(request)
        val body = response.body()
        val code = response.code()
        val headers = response.headers()
        val strBody = if (body == null) "" else body.string()
        printResponse(strBody, headers, code)
        val newBody = ResponseBody.create(body?.contentType(), strBody)
        return response.newBuilder().body(newBody).build()
    }

    private fun printResponse(body: String, headers: Headers, code: Int) {
        Logger.d(TAG, "response code:$code")
        Logger.d(TAG, "body:$body")
        Logger.d(TAG, "headers:$headers")
    }

    private fun printRequest(request: Request) {
        try {
            Logger.d(TAG, "url:" + request.url())
            Logger.d(TAG, "method:" + request.method())
            Logger.d(TAG, "headers:" + request.headers())
            val buffer = Buffer()
            val body = request.body()
            if (body != null) {
                body.writeTo(buffer)
                Logger.d(TAG, "body:" + buffer.readByteString().utf8())
            }
        } catch (e: Exception) {
            Logger.e(TAG, "printRequest", e)
        }

    }
}